docs/cookbook/deploy-using-bentoml.md

```diff
@@ -0,0 +1,201 @@
+# Run Outlines using BentoML
+
+[BentoML](https://bentoml.com/) is an open-source model serving library for building performant and scalable AI applications with Python. It comes with tools that you need for serving optimization, model packaging, and production deployment.
+
+In this guide, we will show you how you can use BentoML to run programs written with Outlines on GPU locally and in the cloud. The example source codes in this guide are also available at [examples/bentoml/](https://github.com/outlines-dev/outlines/blob/main/examples/bentoml/) directory.
+
+## Import model
+
+First we need to download the Mistral-7B-v0.1 model and import the model into BentoML's model store. Let's install the BentoML and other dependencies from PyPi (preferably in a virtual environment):
+
+```bash
+pip install -r requirements.txt
+```
+
+Then save the code snippet below as `import_model.py` and run `python import_model.py`
+
+```python
+import bentoml
+
+MODEL_ID = "mistralai/Mistral-7B-v0.1"
+BENTO_MODEL_TAG = MODEL_ID.lower().replace("/", "--")
+
+def import_model(model_id, bento_model_tag):
+
+    import torch
+    from transformers import AutoModelForCausalLM, AutoTokenizer
+
+    tokenizer = AutoTokenizer.from_pretrained(MODEL_ID)
+    model = AutoModelForCausalLM.from_pretrained(
+        MODEL_ID,
+        torch_dtype=torch.float16,
+        low_cpu_mem_usage=True,
+    )
+
+    with bentoml.models.create(bento_model_tag) as bento_model_ref:
+        tokenizer.save_pretrained(bento_model_ref.path)
+        model.save_pretrained(bento_model_ref.path)
+
+
+if __name__ == "__main__":
+    import_model(MODEL_ID, BENTO_MODEL_TAG)
+```
+
```

Comment:
Perhaps add a result here to let users verify the model is locally available.

```suggestion

You can verify the download is successful by running:

```bash
$ bentoml models list

Tag                                          Module  Size        Creation Time
mistralai--mistral-7b-v0.1:m7lmf5ac2cmubnnz          13.49 GiB   2024-04-25 06:52:39
```
```